components/brave_wallet/renderer/brave_wallet_js_handler.cc

```diff
@@ -248,30 +263,48 @@ void BraveWalletJSHandler::InjectInitScript() {
 }
 
 void BraveWalletJSHandler::FireEvent(const std::string& event,
-                                     const std::string& event_args) {
+                                     base::Value event_args) {
   base::Value args = base::Value(base::Value::Type::LIST);
   args.Append(event);
-  args.Append(event_args);
+  args.Append(std::move(event_args));
   CallMethodOfObject(render_frame_->GetWebFrame(),
                      u"ethereum",
                      u"emit",
                      std::move(args));
 }
 
-void BraveWalletJSHandler::ConnectEvent(const std::string& chain_id) {
-  FireEvent(kConnectEvent, chain_id);
+void BraveWalletJSHandler::ConnectEvent() {
+  if (!EnsureConnected())
+    return;
+
+  brave_wallet_provider_->GetChainId(base::BindOnce(
+      &BraveWalletJSHandler::OnGetChainId, base::Unretained(this)));
+}
+
+void BraveWalletJSHandler::OnGetChainId(const std::string& chain_id) {
```

Comment:
I think it would be helpful to link to the docs for these methods https://web3js.readthedocs.io/en/v1.2.4/web3-eth.html#getchainid

Reply:
the whole implementation follow EIP-1193. I can put it in the header. But the whole web3 provider is following a standard for web3 providers

Reply:
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md